src/Zenith.sol

```diff
@@ -67,12 +73,36 @@ contract Zenith is Passage {
     /// @notice Emit the entire block data for easy visibility
     event BlockData(bytes blockData);
 
-    /// @notice Initializes the Admin role.
-    /// @dev See `AccessControlDefaultAdminRules` for information on contract administration.
-    ///      - Admin role can grant and revoke Sequencer roles.
-    ///      - Admin role can be transferred via two-step process with a 1 day timelock.
-    /// @param admin - the address that will be the initial admin.
-    constructor(uint256 defaultRollupChainId, address admin) Passage(defaultRollupChainId, admin) {}
+    /// @notice Emitted when a sequencer is added or removed.
+    event SequencerSet(address sequencer, bool permissioned);
+
+    constructor(uint256 _defaultRollupChainId, address _withdrawalAdmin, address _sequencerAdmin)
+        Passage(_defaultRollupChainId, _withdrawalAdmin)
+    {
+        sequencerAdmin = _sequencerAdmin;
+    }
+
+    /// @notice Add a sequencer to the permissioned sequencer list.
+    /// @param sequencer - the address of the sequencer to add.
+    /// @custom:emits SequencerSet if the sequencer is added.
+    /// @custom:reverts OnlySequencerAdmin if the caller is not the sequencerAdmin.
+    function addSequencer(address sequencer) external {
+        if (msg.sender != sequencerAdmin) revert OnlySequencerAdmin();
+        if (isSequencer[sequencer]) return;
```

Comment:
probably unnecessary and may add an extra SLOAD

Reply:
ditto above